core/opendaq/reader/src/stream_reader_impl.cpp

```diff
@@ -548,6 +565,27 @@ ErrCode StreamReaderImpl::readWithDomain(void* samples,
     return errCode;
 }
 
+ErrCode StreamReaderImpl::skipSamples(SizeT* count)
+{
+    OPENDAQ_PARAM_NOT_NULL(count);
+
+    std::scoped_lock lock(mutex);
+
+    if (invalid)
+        return makeErrorInfo(OPENDAQ_ERR_INVALID_DATA, "Invalid reader state", nullptr);
+
+    ErrCode errCode = OPENDAQ_SUCCESS;
+    info.prepare(nullptr, *count, milliseconds(0));
+    if (info.dataPacket.assigned())
+        errCode = readPacketData();
+
+    if (OPENDAQ_SUCCEEDED(errCode) && info.remainingToRead <= *count)
+        errCode = readPackets(nullptr);
+
+    *count = *count - info.remainingToRead;
+    return errCode;
+}
```

Comment:
This must return `IReaderStatus`, or you will end up dropping events, making them irrecoverable.